feathr_project/feathr/spark_provider/_localspark_submission.py

```diff
@@ -0,0 +1,224 @@
+            main_jar_path (str): main file paths, usually your main jar file
+            main_class_name (str): name of your main class
+            arguments (str): all the arguments you want to pass into the spark job
+            configuration (Dict[str, str]): Additional configs for the spark job
+            python_files (List[str]): required .zip, .egg, or .py files of spark job
+            properties (Dict[str, str]): Additional System Properties for the spark job
+            job_tags (str): not used in local spark mode
+            reference_files_path (str): not used in local spark mode
+        """
+        logger.warning(f"Local Spark Mode only support basic params right now and should be used only for testing purpose.")
+        self.cmd_file, self.log_path = self._get_debug_file_name(self.debug_folder, prefix = job_name)
+        args = self._init_args(job_name)
+
+        if properties:
+            arguments.append("--system-properties %s" % json.dumps(properties))
+
+        if configuration:
+            cfg = configuration.copy()  # We don't want to mess up input parameters
+        else:
+            cfg = {}
+        
+        if not main_jar_path:
+            # We don't have the main jar, use Maven
+            # Add Maven dependency to the job configuration
+            if "spark.jars.packages" in cfg:
+                cfg["spark.jars.packages"] = ",".join(
+                    [cfg["spark.jars.packages"], FEATHR_MAVEN_ARTIFACT])
+            else:
+                default_packages = "org.apache.spark:spark-avro_2.12:3.3.0,com.microsoft.sqlserver:mssql-jdbc:10.2.0.jre8,com.microsoft.azure:spark-mssql-connector_2.12:1.2.0,org.apache.logging.log4j:log4j-core:2.17.2,com.typesafe:config:1.3.4,com.fasterxml.jackson.core:jackson-databind:2.12.6.1,org.apache.hadoop:hadoop-mapreduce-client-core:2.7.7,org.apache.hadoop:hadoop-common:2.7.7,org.apache.avro:avro:1.8.2,org.apache.xbean:xbean-asm6-shaded:4.10,org.apache.spark:spark-sql-kafka-0-10_2.12:3.1.3,com.microsoft.azure:azure-eventhubs-spark_2.12:2.3.21,org.apache.kafka:kafka-clients:3.1.0,com.google.guava:guava:31.1-jre,it.unimi.dsi:fastutil:8.1.1,org.mvel:mvel2:2.2.8.Final,com.fasterxml.jackson.module:jackson-module-scala_2.12:2.13.3,com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.12.6,com.fasterxml.jackson.dataformat:jackson-dataformat-csv:2.12.6,com.jasonclawson:jackson-dataformat-hocon:1.1.0,com.redislabs:spark-redis_2.12:3.1.0,org.apache.xbean:xbean-asm6-shaded:4.10,com.google.protobuf:protobuf-java:3.19.4,net.snowflake:snowflake-jdbc:3.13.18,net.snowflake:spark-snowflake_2.12:2.10.0-spark_3.2,org.apache.commons:commons-lang3:3.12.0,org.xerial:sqlite-jdbc:3.36.0.3,com.github.changvvb:jackson-module-caseclass_2.12:1.1.1,com.azure.cosmos.spark:azure-cosmos-spark_3-1_2-12:4.11.1,org.eclipse.jetty:jetty-util:9.3.24.v20180605,commons-io:commons-io:2.6,org.apache.hadoop:hadoop-azure:2.7.4,com.microsoft.azure:azure-storage:8.6.4"
```

Comment:
Break down this line, it's toooooooooo long.

Reply:
put them in a separate function with todo items to refine with dynamic modulation in the future.